jobmon_core/src/jobmon/core/requester.py

```diff
@@ -22,165 +24,124 @@ def http_request_ok(status_code: int) -> bool:
+        return cls(
+            service_url, request_timeout, retries_timeout, retries_attempts, use_otlp
+        )
 
     def add_server_structlog_context(self, **kwargs: Any) -> None:
         """Add the structlogging context if it has been provided."""
         for key, value in kwargs.items():
             self.server_structlog_context[key] = value
 
-    def send_request(
-        self,
-        app_route: str,
-        message: dict,
-        request_type: str,
-        tenacious: bool = True,
-    ) -> Tuple[int, Any]:
-        """Send request to server.
-
-        If we get a 5XX status code, we will retry for up to 2 minutes using
-        exponential backoff.
-
-        Args:
-            app_route:
-                The specific end point with which you want to
-                interact. The app_route must always start with a slash ('/') and
-                must match one of the function decorations of @jsm.route or
-                @jqs.route on the server side.
-            message: The message dict to be sent to the server.
-                Must contain any arguments the JSM/JQS route needs to operate.
-                If the request is a 'GET', the value of the message dict will
-                likely be parsed into the url. If the request is a 'POST' or 'PUT',
-                the message dict will get stored in a dictionary that is parsed on
-                the server side and passed into the work done by that route.
-                For example, a valid message for a request to add a task_dag
-                to the JSM might be:
-
-                    {'name': 'my_name',
-                     'user': 'my_user',
-                     'dag_hash': 'my_dag_hash'}
-
-            request_type: The type of request desired, either 'get', 'post, or 'put'
-            tenacious: use tenacity for retries
-
-
-        Returns:
-            Server reply message
-
-        Raises:
-            RuntimeError if 500 errors occur for > 2 minutes
-
-        """
-        if tenacious:
-            res = self._tenacious_send_request(app_route, message, request_type)
+    @contextlib.contextmanager
+    def tracing_span(self, app_route, request_type):
+        if self._otlp_api:
+            tracer = self._otlp_api.get_tracer("requester")
+            with tracer.start_as_current_span("send_request") as span:
+                span.set_attribute("http.method", request_type.upper())
+                span.set_attribute("http.url", self.url + app_route)
+                yield span
         else:
-            res = self._send_request(app_route, message, request_type)
-
-        if not http_request_ok(res[0]):
-            raise InvalidResponse(
-                f"Unexpected status code {res[0]} from POST "
-                f"request through route {app_route}. Expected "
-                f"code 200. Response content: {res[1]}"
+            yield None
+
+    def _maybe_trace(self, func):
+        @functools.wraps(func)
```

Comment:
kind of a nitpick but feels like the "maybe" part should be checked here, i.e. `if not self._otlp_api: return func`instead of running inside a null context. This would match the behavior of `_maybe_retry`.